### PR TITLE
Reformat cmake files

### PR DESCRIPTION
**Pull request by cgmb**
There are a few manual touchups for indentation levels, but this is mostly just running:

    sed -Ei \
        -e 's/\( /\(/g' \
        -e 's/([^[:space:]]+) \)/\1\)/g' \
        -e 's/(if|elif|endif|set|list|file) \(/\1(/g' \
        -e 's/#(\w)/# \1/'

These transformations eliminate spacing after opening parentheses, before closing parentheses (if there was any text on the same line), between the keyword and the opening parentheses, and ensure there is a space after comment markers.

### CODE REVIEW

File: library/src/CMakeLists.txt [hunk 1]
```diff
@@ -22,26 +22,26 @@
 
 
 # ########################################################################
-# A helper function to prefix a source list of files with a common path into a new list (non-destructive)
+# A helper function to prefix a source list of files with a common path into a new list(non-destructive)
```
Comment (tfalders) [inline library/src/CMakeLists.txt]: Looks like this comment got caught in the automatic replacement

**Review by tfalders — APPROVED** [review]
Awesome! Thanks a bunch for this

File: library/src/CMakeLists.txt (hunk 1 above)
Comment (cgmb) [reply library/src/CMakeLists.txt]: Good catch.

**Review by jmachado-amd — APPROVED** [review]
That is quite an aesthetic improvement, thanks for doing it Cory!

**Pull request merged**